### PR TITLE
Adding support for local alignment in NeedlemanWunschAligner

### PR DESCRIPTION
**Pull request by nh13**
Allows a subsequence of the query to be aligned to a subsequence of the
target.

Fixed a few spelling typos in the NeedlemanWunschAlignerTest.

### CODE REVIEW

File: src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala [hunk 1]
```diff
@@ -278,10 +286,94 @@ class NeedlemanWunschAlignerTest extends UnitSpec {
...
+    assertValidLocalAlignment(result)
+    result.queryStart shouldBe 3
+    result.targetStart shouldBe 1
+    result.queryEnd shouldBe 13
+    result.targetEnd shouldBe 10
+    result.cigar.toString() shouldBe "5=1I5="
+    result.score shouldBe 6
+  }
+
+  it should "correctly align prefer a match over an indel" in {
```
Comment (nh13) [inline src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala]: this is implementation specific, where we check the max score is from the diagonal before up/down.

File: src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala [hunk 2]
```diff
@@ -24,29 +24,37 @@
...
-import com.fulcrumgenomics.alignment.Mode.Glocal
+import com.fulcrumgenomics.alignment.Mode.{Glocal, Local}
 import com.fulcrumgenomics.testing.UnitSpec
 
 class NeedlemanWunschAlignerTest extends UnitSpec {
   /** Upper-cases and remove display-related characters from a string. */
   def s(str: String): String = str.filterNot(ch => ch == '-').toUpperCase
 
-  /** Asserts a couple of things about an alignment that should be constant for global alignments. */
+  /** Asserts a couple of things about alignment that should be constant for global alignments. */
```
Comment (tfenne) [inline src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala]: I think this read better before.  Alignment needs an article before it, either `an` or `the`.

File: src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala [hunk 3]
```diff
@@ -24,29 +24,37 @@
...
     alignment.targetStart should be >= 1
     alignment.targetEnd   should be <= alignment.target.length
   }
 
+  /** Asserts a couple of things about alignment that should be constant for local alignments. */
+  def assertValidLocalAlignment(alignment: Alignment): Unit = {
+    alignment.queryStart should be >= 1
+    alignment.queryEnd   should be <= alignment.query.length
+    alignment.targetStart should be >= 1
+    alignment.targetEnd   should be <= alignment.target.length
```
Comment (tfenne) [inline src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala]: Do you want to also assert that score >= 0 for a local alignment?

File: src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala [hunk 4]
```diff
@@ -278,10 +286,94 @@ class NeedlemanWunschAlignerTest extends UnitSpec {
     result.score shouldBe 4
   }
 
-  it should "align a query that is longer than the target, creating an insertino" in {
```
Comment (tfenne) [inline src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala]: `insertino` it's Italian for insertion 😛

File: src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala (hunk 1 above)
Comment (tfenne) [reply src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala]: Sounds good.  The name of the test is a little off though, perhaps just remove "correctly align".

**Pull request merged**